src/main/java/store/domain/promotion/Promotion.java

```diff
@@ -0,0 +1,58 @@
+package store.domain.promotion;
+
+import store.domain.promotionCondition.PromotionCondition;
+
+public class Promotion {
+    private final String promotionName;
+    private final int purchaseQuantity;
+    private final int giftQuantity;
+    private final PromotionCondition promotionCondition;
+
+    public Promotion(String promotionName, int purchaseQuantity, int giftQuantity, PromotionCondition promotionCondition) {
+        this.promotionName = promotionName;
+        this.purchaseQuantity = purchaseQuantity;
+        this.giftQuantity = giftQuantity;
+        this.promotionCondition = promotionCondition;
+    }
+
+    public boolean isActive() {
+        return promotionCondition.isSatisfiedBy();
+    }
+    public int calculateGiftQuantity(int quantity) {
+        if (!isActive()) {
+            return 0;
+        }
+
+        int promotionUnit = purchaseQuantity + giftQuantity;
+        return (quantity / promotionUnit) * giftQuantity;
+    }
+
+    public int calculateAdditionalGiftQuantity(int quantity) {
+        if (!isActive()) {
+            return 0;
+        }
+
+        int promotionUnit = purchaseQuantity + giftQuantity;
+        int additionalQuantityNeeded = promotionUnit - (quantity % promotionUnit);
+
+        if (additionalQuantityNeeded <= giftQuantity) {
+            return additionalQuantityNeeded;
+        }
+        return 0;
+
+    }
```

Comment:
promotionUnit - (x % 3) == giftQuantity 는 결국,
promotionUnit - giftQuantity == x % 3 이 되므로, 다음과 같이 간단하게 작성할 수 있을 것 같습니다.

```suggestion
    public int calculateAdditionalGiftQuantity(int quantity) {
        if (!isActive()) {
            return 0;
        }

        if (quantity % (purchaseQuantity + giftQuantity) == purchaseQuantity) {
            return giftQuantity;
        }
        return 0;

    }
```